bigframes/core/log_adapter.py

```diff
@@ -0,0 +1,38 @@
+import functools
+
+from loguru import logger
+
+_log_file_path = None
+_logger = logger
+_api_methods = []
```

Comment:
I'm confused by this. Wouldn't we want to store the `_api_methods` on the individual objects. What happens when we have two different objects? I don't think we want to log all methods ever in the API request, just the ones used to generate the object.